[BUG] Env Injector fails on pods with generateName

**Components and versions**
Select which component(s) the bug relates to with [X].

[ ] Controller, version: `x.x.x` (docker image tag)
[X] Env-Injector (webhook), version: `1.3.0` (docker image tag)
[ ] Other

**Describe the bug**
I'm trying to inject secrets into a pod that has an autogenerated name. This pod does not have a name or ownerReference set. I believe it is failing because when creating a secret for it the secret name is invalid. See code here: https://github.com/SparebankenVest/azure-key-vault-to-kubernetes/blob/bb751aef3405ab1b0118576f73635667e8481a10/cmd/azure-keyvault-secrets-webhook/auth/auth.go#L251

```go
	name := pod.GetName()
	ownerReferences := pod.GetOwnerReferences()
	if name == "" {
		if len(ownerReferences) > 0 {
			if strings.Contains(ownerReferences[0].Name, "-") {
				generateNameSlice := strings.Split(ownerReferences[0].Name, "-")
				name = strings.Join(generateNameSlice[:len(generateNameSlice)-1], "-")
			} else {
				name = ownerReferences[0].Name
			}
		}
	}
```

**To Reproduce**
Attempt to inject a secret into this pod:
```yaml
apiVersion: v1
kind: Pod
metadata:
  generateName: pod-inject
  namespace: akv-test
spec:
  containers:
    - name: akv2k8s-env-test
      image: spvest/akv2k8s-env-test:2.0.1
      args: ["TEST_SECRET", "TEST_HARDCODED"]
      env:
        - name: TEST_SECRET
          value: "my-secret@azurekeyvault" # ref to akvs
        - name: TEST_HARDCODED
          value: "blah"
```
`kubectl create -f pod-inject.yaml`

**Expected behavior**
The secret should be injected into the image without errors.

**Logs**
If applicable, add logs to help explain your problem.

```
I0114 00:44:03.470012       1 pod.go:292] "mutate init-containers" akv-test/="(MISSING)"
I0114 00:44:03.470052       1 pod.go:298] "mutate containers" akv-test/="(MISSING)"
I0114 00:44:03.470064       1 pod.go:116] "found container to mutate" container="akv-test/akv2k8s-env-test"
I0114 00:44:03.470075       1 pod.go:119] "checking for env vars to inject" container="akv-test/akv2k8s-env-test"
I0114 00:44:03.470097       1 pod.go:122] "found env var to inject" env="configuration-management-username@azurekeyvault" container="akv-test/akv2k8s-env-test"
I0114 00:44:03.470112       1 registry.go:30] "getting container command for container" container="akv-test/akv2k8s-env-test"
I0114 00:44:03.470135       1 registry.go:36] "no cmd override in kubernetes for container, checking docker image configuration for entrypoint and cmd" image="spvest/akv2k8s-env-test:2.0.1" container="akv-test/akv2k8s-env-test"
I0114 00:44:03.493495       1 round_trippers.go:454] GET https://10.220.128.1:443/api/v1/namespaces/akv-test/serviceaccounts/default 200 OK in 23 milliseconds
E0114 00:44:03.645841       1 aws_credentials.go:77] while getting AWS credentials NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
I0114 00:44:04.628595       1 pod.go:149] "found container arguments to use for env-injector" cmd="entrypoint.sh TEST_SECRET TEST_HARDCODED" container="akv-test/akv2k8s-env-test"
I0114 00:44:04.807339       1 pod.go:265] "signed arguments to prevent override" container="akv-test/akv2k8s-env-test"
I0114 00:44:04.807432       1 pod.go:272] "public signing key for argument verification" key="..." container="akv-test/akv2k8s-env-test"
I0114 00:44:04.807452       1 pod.go:159] "full exec path" path="/azure-keyvault/azure-keyvault-env" container="akv-test/akv2k8s-env-test"
I0114 00:44:04.807471       1 pod.go:170] "mounting volume" volume="azure-keyvault-env" path="/azure-keyvault/" container="akv-test/akv2k8s-env-test"
I0114 00:44:04.809841       1 round_trippers.go:454] POST https://xx.xx.xx.xx.1:443/api/v1/namespaces/akv-test/secrets 422 Unprocessable Entity in 2 milliseconds
E0114 00:44:04.810169       1 main.go:166] "failed to mutate" err="Secret \"akv2k8s-\" is invalid: metadata.name: Invalid value: \"akv2k8s-\": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')" pod="akv-test/"
2022/01/14 00:44:04 [ERROR] admission webhook error: Secret "akv2k8s-" is invalid: metadata.name: Invalid value: "akv2k8s-": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

**Additional context**
Add any other context about the problem here.

--- a/cmd/azure-keyvault-secrets-webhook/auth/auth.go
+++ b/cmd/azure-keyvault-secrets-webhook/auth/auth.go
@@ -272,6 +272,8 @@ func (a AuthService) NewPodSecret(pod *corev1.Pod, namespace string, mutationID 
 			} else {
 				name = ownerReferences[0].Name
 			}
+		} else {
+			name = pod.GetGenerateName()
 		}
 	}
 

